src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,333 @@
+import {DEFAULT_STEP, FIRST_POINT, MINIMUM_TRACK_WIDTH, defaultHitSlop} from './Constants'
+import {Thumb, TrackPoint} from './components'
+
+type Size = {
+  width: number
+  height: number
+}
+
+type SliderInfo = {
+  range: number
+  trackWidth: number
+}
+
+type AnimatedGHContext = {
+  startX: number
+  currentPoint: number
+}
+
+type ThumbContainerStyle = {
+  thumbSize: Size
+  hasThumbComponent?: boolean
+}
+
+type TrackStyle = {
+  backgroundColor: string
+}
+
+type TrackPointStyle = {
+  width: number
+}
+
+interface AnimatedLabelProps extends TextInputProps {
+  text: string
+}
+
+type FlexDirection = 'column' | 'column-reverse' | 'row' | 'row-reverse'
+type Position = 'absolute' | 'fixed' | 'relative' | 'static' | 'sticky'
+
+export interface SliderCommonProps {
+  /** The maximum value of the slider */
+  maximumValue?: number
+
+  /** The minimum value of the slider */
+  minimumValue?: number
+
+  /** The step value for the slider */
+  step?: number
+
+  /** The alwaysShowValue indicates whether the value of the slider should always be displayed */
+  alwaysShowValue?: boolean
+
+  /** Whether to show the point on the slider's track */
+  hasTrackPoint?: boolean
+
+  /** Determines whether the thumb can be moved by directly touching the thumb or only by dragging the slider track */
+  hasPointTouch?: boolean
+
+  /** The touchable area is used to increase the size of the thumb and make it easier to interact with */
+  hitSlopPoint?: Insets | number
+
+  /** Style of the slider's track */
+  trackStyle?: StyleProp<ViewStyle>
+
+  /** The bgColorTrack sets the background color of the slider's track */
+  bgColorTrack?: string
+
+  /** The bgColorTracked sets the background color of the part of the track that has been selected */
+  bgColorTracked?: string
+
+  /** Style of the point on the slider's track */
+  trackPointStyle?: StyleProp<ViewStyle>
+
+  /** The bgColorLabelView sets the background color of the view that displays the value of the slider */
+  bgColorLabelView?: string
+
+  /** The labelStyle sets the style of the text that displays the value of the slider */
+  labelStyle?: StyleProp<TextProps>
+
+  /** Style of the slider's thumb */
+  thumbStyle?: StyleProp<ViewStyle>
+
+  /** custom element that can be used to replace the default thumb of the slider */
+  thumbComponent?: React.ReactElement
+
+  /** Size of the slider's thumb */
+  thumbSize?: Size
+
+  /** Callback function to handle the change in slider value */
+  onValueChange?: (value: number | {minimum: number; maximum: number}) => void
+}
+
+type SliderPropsWithOptionalWidth = {sliderWidth?: number} & (
+  | {hasTrackPoint: true; sliderWidth: number}
+  | {hasTrackPoint?: false; sliderWidth?: number}
+)
+
+type SliderProps = SliderCommonProps & SliderPropsWithOptionalWidth
+
+type SliderComponentProps = React.FunctionComponent<SliderProps> & {
+  Range: React.FunctionComponent<SliderProps>
+}
+
+const Slider: SliderComponentProps = ({
```

Comment:
const Slider : React.FC< SliderComponentProps > = () => ({})

Reply:
I changed 'React.FunctionalComponent' to 'React.FC', but now I need to use '&' for the compound component

type SliderComponentProps = React.FC<SliderProps> & {
  Range: React.FC<ISliderRange>
}